### DISCUSSION TITLE
Error: oak9cli: Unable to get versions

### DISCUSSION THREAD

**Discussion by awright-oak9** [Homebrew/discussions#4224] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 4 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 69deebcfe1d2d6f37cb3f3ebb08d9f2ad305d006
Core tap last commit: 4 weeks ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hello, I'm trying to submit a  cask to homebrew but when running brew livecheck I'm getting Error: oak9cli: Unable to get versions

brew livecheck --debug oak9cli

Cask:             oak9cli
Livecheckable?:   Yes

URL:              https://downloads.oak9.io/#oak9cli/v0.11.14/
Strategy:         PageMatch
Regex:            /oak9cli-osx-v(\d+(?:\.\d+)*)\.zip/i
Error: oak9cli: Unable to get versions

Cask  file:

cask "oak9cli" do
  version "0.11.14"
  sha256 "b9568ed466d16c5b67fa081856defb6cc77ca44fb543a7e9f330fafbd04b276e"

  url "https://downloads.oak9.io/oak9cli%2Fv#{version}%2Foak9cli-osx-v#{version}.zip"
  name "oak9cli"
  desc "test desc"
  homepage "https://oak9.io/"

  app "oak9cli"

  livecheck do
    url "https://downloads.oak9.io/#oak9cli/v#{version}/"
    regex(/oak9cli-osx-v(\d+(?:\.\d+)*)\.zip/i)
  end
end

Can someone please point me in the right direction here?

Thanks!

**Comment by p-linnane** [comment]
The page you are checking is using JavaScript. `livecheck` uses `curl` to grab the page to parse, so it can't handle JS.

**Reply by awright-oak9** [reply]
I  removed the javascript and updated the livecheck to the below but still getting the same error.

livecheck do
    url "https://downloads.oak9.io/oak9cli/v#{version}/"
    regex(/oak9cli-osx-v(\d+(?:\.\d+)*)\.zip/i)
  end

brew livecheck --debug oak9cli

Cask:             oak9cli
Livecheckable?:   Yes

URL:              https://downloads.oak9.io/oak9cli/v0.11.14/
Strategy:         PageMatch
Regex:            /oak9cli-osx-v(\d+(?:\.\d+)*)\.zip/i
Error: oak9cli: Unable to get versions

Thanks

**Reply by gromgit** [reply]
https://downloads.oak9.io/oak9cli/v0.11.14/ now returns nothing, literally.

**Reply by p-linnane** [reply]
Additionally, `https://downloads.oak9.io/oak9cli/v#{version}/` will always return the existing version. This needs to be a URL that is static.

**Reply by awright-oak9** [reply]
without the javascript there is no directory indexing so you have to hit the file directly EX: curl  https://downloads.oak9.io/oak9cli/v0.11.14/oak9cli-osx-v0.11.14.zip -o oak9cli.zip

**Reply by awright-oak9** [reply]
@p-linnane are you saying I can't store each version in its own directory and have the pipeline update the cask file every time we publish a new one?

**Reply by p-linnane** [reply]
You can use a directory root and have `livecheck` pull the latest version from the results.

**Reply by awright-oak9** [reply]
So pulling the version from the filename or directory wont work?

**Reply by p-linnane** [reply]
You can use either of those. What I mean is that the link would have to be something like `https://downloads.oak9.io/oak9cli/latest`. `livecheck` could use that link and fetch the filename from the headers. Another option is to have a static page with version information that `livecheck` can scan the page for.

**Reply by p-linnane** [reply]
An example is `bash`. We use `livecheck` to scan this directory root and find the highest version number: https://ftp.gnu.org/gnu/bash/

This could be the files themselves, folders, etc.
